src/Http/Routing/test/UnitTests/Builder/MinimalActionEndpointRouteBuilderExtensionsTest.cs

```diff
@@ -359,6 +359,84 @@ public void MapFallbackWithoutPath_BuildsEndpointWithLowestRouteOrder()
+            Assert.Equal(name, endpointName?.EndpointName);
+            Assert.Equal(name, routeName?.RouteName);
+            Assert.Equal("HTTP: DELETE / => InnerGetString", routeEndpointBuilder.DisplayName);
+        }
+
+        [Fact]
+        public void MapMethod_SetsEndpointNameForMethodGroup()
+        {
+            var name = "GetString";
+            var builder = new DefaultEndpointRouteBuilder(new ApplicationBuilder(new EmptyServiceProvdier()));
+            _ = builder.MapDelete("/", GetString);
+
+            var dataSource = GetBuilderEndpointDataSource(builder);
+            // Trigger Endpoint build by calling getter.
+            var endpoint = Assert.Single(dataSource.Endpoints);
+
+            var endpointName = endpoint.Metadata.GetMetadata<IEndpointNameMetadata>();
+            var routeName = endpoint.Metadata.GetMetadata<IRouteNameMetadata>();
+            var routeEndpointBuilder = GetRouteEndpointBuilder(builder);
+            Assert.Equal(name, endpointName?.EndpointName);
+            Assert.Equal(name, routeName?.RouteName);
+            Assert.Equal("HTTP: DELETE / => GetString", routeEndpointBuilder.DisplayName);
+        }
+
+        [Fact]
+        public void WithNameOverridesDefaultEndpointName()
+        {
+            var name = "SomeCustomName";
+            var builder = new DefaultEndpointRouteBuilder(new ApplicationBuilder(new EmptyServiceProvdier()));
+            _ = builder.MapDelete("/", GetString).WithName(name);
+
+            var dataSource = GetBuilderEndpointDataSource(builder);
+            // Trigger Endpoint build by calling getter.
+            var endpoint = Assert.Single(dataSource.Endpoints);
+
+            var endpointName = endpoint.Metadata.GetMetadata<IEndpointNameMetadata>();
+            var routeName = endpoint.Metadata.GetMetadata<IRouteNameMetadata>();
+            var routeEndpointBuilder = GetRouteEndpointBuilder(builder);
+            Assert.Equal(name, endpointName?.EndpointName);
+            Assert.Equal(name, routeName?.RouteName);
+            // Will still use the original method name, not the custom endpoint name
+            Assert.Equal("HTTP: DELETE / => GetString", routeEndpointBuilder.DisplayName);
+        }
+
+        private string GetString() => "TestString";
+
+        [Fact]
+        public void MapMethod_DoesNotSetEndpointNameForLambda()
+        {
+            var builder = new DefaultEndpointRouteBuilder(new ApplicationBuilder(new EmptyServiceProvdier()));
+            _ = builder.MapDelete("/", () => { });
+
+            var dataSource = GetBuilderEndpointDataSource(builder);
+            // Trigger Endpoint build by calling getter.
+            var endpoint = Assert.Single(dataSource.Endpoints);
+
+            var endpointName = endpoint.Metadata.GetMetadata<IEndpointNameMetadata>();
+            Assert.Null(endpointName);
+        }
+
         class FromRoute : Attribute, IFromRouteMetadata
```

Comment:
Add a test for a method on a class.

Reply:
Isn't `MapMethod_SetsEndpointNameForMethodGroup` sufficient for that?

Reply:
It occurs to me that all methods have to exist on classes 😄. Yes, you're right.